lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4338,6 +4340,35 @@ namespace {
             return result;
         }
       }
+      if (auto importingContext = Impl.getImportingContext()) {
+        llvm::SmallVector<ValueDecl *> results;
+        llvm::SmallVector<ImportedModule> importedModules;
+
+        const ModuleDecl *importingSwiftModule =
+            importingContext.getValue()->getParentModule();
+
+        if (importingSwiftModule->isNonSwiftModule())
+          return nullptr;
+
+        const SourceFile *importingSwiftSourceFile =
+            importingContext.getValue()->getParentSourceFile();
+
+        // TODO: Make sure we aren't bailing early in legitimate cases
+        if (!importingSwiftSourceFile)
```

Comment:
It seems there are cases where the importing DeclContext does not lie within a source file. I would've thought when the main module is already serialized and being inspected by swift-ide-test would be such a case, but it doesn't seem so. As the comment says, lets make sure by bailing here we aren't leaving something important on the table.